### PR TITLE
Add integration for QtQuick Controls 2 applications

### LINKED ISSUE #129: Integration for QML applications
Qt Quick Controls 2 applications use the `Default` style if no other style is set, and that one is really basic  and sometimes not usable.

It would be nice if QGnomePlatform would set this to `org.kde.desktop`, which just passes through the widget style (for example Adwaita-Qt).

A simply copy-paste from [Plasma Integration](https://invent.kde.org/plasma/plasma-integration/-/blob/master/src/platformtheme/kdeplatformtheme.cpp#L578-L581) should be enough.

### PR DESCRIPTION
**Pull request by JBBgameich**
~This is untested~, but should work

Fixes #129

### CODE REVIEW

**Comment by grulja** [comment]
I think the dependency on qqc2-desktop-style should be runtime and optional, because we don't want to automatically bring KDE Frameworks into GNOME. There should be some runtime check if qqc2-desktop-style is available and use it in such case. I would then expect to use something like `Recommends: qqc2-desktop-style` in Fedora packaging where we can suggest this as an optional dependency.

**Comment by JBBgameich** [comment]
I now changed it to only setting the style when it is available.
Personally I disagree with the prioritization of avoiding KF5, since applications are really broken without the style, and it has a really small dependency tree.
```
qml-module-org-kde-qqc2desktopstyle
  Depends: qml-module-org-kde-kirigami2
  Depends: qml-module-org-kde-sonnet
  Depends: qml-module-qtgraphicaleffects
  Depends: qml-module-qtqml-models2
  Depends: qml-module-qtquick-controls2
  Depends: qml-module-qtquick-layouts
  Depends: qml-module-qtquick-templates2
  Depends: qml-module-qtquick-window2
  Depends: qml-module-qtquick2
  Depends: libc6
  Depends: libkf5configcore5
  Depends: libkf5configwidgets5
  Depends: libkf5iconthemes5
  Depends: libkf5kirigami2-5
  Depends: libqt5core5a
  Depends: libqt5dbus5
 |Depends: libqt5gui5
  Depends: libqt5gui5-gles
  Depends: libqt5qml5
 |Depends: libqt5quick5
  Depends: libqt5quick5-gles
  Depends: libqt5widgets5
  Depends: libstdc++6
```

My motivation for doing this was https://mastodon.social/@tbernard/109897404484985777 btw, this is also a good example for how things look without the desktop style.

File: src/common/gnomesettings.h [hunk 1]
```diff
@@ -23,11 +23,14 @@
 #include <QFlags>
 #include <QObject>
 #include <QStringList>
+#include <QLoggingCategory>
 
 #include <qpa/qplatformtheme.h>
 
 #include <memory>
 
+Q_DECLARE_LOGGING_CATEGORY(QGnomePlatform);
```
Comment (grulja) [inline src/common/gnomesettings.h:32]: This should be something like what we use in `common/`:
e.g. `Q_LOGGING_CATEGORY(QGnomePlatformTheme, "qt.qpa.qgnomeplatform.theme")`

**Review by grulja — CHANGES_REQUESTED** [review]

**Comment by grulja** [comment]
Btw. it also doesn't build with Qt6.

**Comment by JBBgameich** [comment]
Should be all fixed now :)

File: src/theme/qgnomeplatformtheme.cpp [hunk 2]
```diff
@@ -24,7 +24,9 @@
 #include "qxdgdesktopportalfiledialog_p.h"
 
 #include <QApplication>
+#include <QDebug>
```
Comment (grulja) [inline src/theme/qgnomeplatformtheme.cpp:27]: Include QLoggingCategory instead.

**Review by grulja — CHANGES_REQUESTED** [review]
Looks good to me. Just fix the include and it's good to go. Thank you.

File: src/theme/qgnomeplatformtheme.cpp (hunk 2 above)
Comment (grulja) [reply src/theme/qgnomeplatformtheme.cpp:27]: This will also fix the failing build.

**Review by grulja — APPROVED** [review]

**Pull request merged**